cloudmarker/clouds/gcpcloud.py

```diff
@@ -121,32 +119,47 @@ def _get_instances(self):
                 object contains details of a virtual machine instance.
 
         """
+        # Fetch all available zones for the current project
+        get_zones = self._compute_resource.zones()
+        list_zones = get_zones.list(project=self._project_name)
+        project_zones = list_zones.execute()
+
+        # Accumulate all zone names from list of zone metadata objects.
+        zones = [item['name'] for item in project_zones['items']]
+
         # Get instance resource to get the list of instances for a project and
         # execute that request.
         instance_resource = self._compute_resource.instances()
         instances = []
         next_page_token = None
 
-        while True:
-            # Prepare the request to get the list of all VM instances for a
-            # project and execute that request.
-            instances_object = instance_resource.list(
-                project=self._project_name,
-                pageToken=next_page_token,
-                zone=self._zone).execute()
-
-            instances.extend(instances_object['items'])
-
-            # instances_object will only contain the maximum number of results
-            # per page which is specified in options keyword argument
-            # maxResults in list() API call. The default value of maxResults is
-            # 500, if the result contains more than 500 records, nextPageToken
-            # key is present in response whose value is to be used to make the
-            # subsequent list() call.
-            if 'nextPageToken' not in instances_object.keys():
-                break
-
-            next_page_token = instances_object['nextPageToken']
+        # Iterating through all zones in the zones list, and collecting
+        # virtual machine instance details for each zone.
+        for zone in zones:
+            while True:
+                # Prepare the request to get the list of all VM instances for a
+                # project and execute that request.
+                instances_object = instance_resource.list(
```

Comment:
Can you rename this variable to something simple, say, `instance`? Like I mentioned earlier, everything is an object anyway, so having `object` in the name is redundant.

Reply:
This variable name was already a part of the code, and was not introduced by me.

Reply:
Sounds good. I will go ahead and test this pull request and if everything looks good, merge it to `master`.